stacks/spica/packages/material/clipboard/clipboard.directive.ts

```diff
@@ -8,19 +8,31 @@ export class MatClipboardDirective {
   @Input() icon: string = "info";
   @Input("matClipboard") text: string;
 
+  clipBoardElement: HTMLInputElement;
+
   @HostListener("click")
   copy() {
     if (!this.text) {
       return;
     }
-    const clipboard = document.createElement("input");
-    clipboard.innerText = this.text;
-    clipboard.select();
-    clipboard.setSelectionRange(0, this.text.length);
+
+    this.prepareClipBoard(this.text);
```

Comment:
This line can be refactored to `const element = this.prepareClipBoard(this.text);` if the `prepareClipBoard` method can return the newly created  element.